src/test/emulators/auth/misc.spec.ts

```diff
@@ -39,6 +44,31 @@ describeAuthEmulator("token refresh", ({ authApi }) => {
       });
   });
 
+  it("should populate auth_time to match lastLoginAt (in seconds since epoch)", async () => {
+    const emailUser = { email: "alice@example.com", password: "notasecret" };
+    const { refreshToken } = await registerUser(authApi(), emailUser);
+
+    const res = await authApi()
+      .post("/securetoken.googleapis.com/v1/token")
+      .type("form")
+      // snake_case parameters also work, per OAuth 2.0 spec.
+      .send({ refresh_token: refreshToken, grantType: "refresh_token" })
+      .query({ key: "fake-api-key" });
+
+    const idToken = res.body.id_token;
+    const user = await getAccountInfoByIdToken(authApi(), idToken);
+    const lastLoginAtSeconds = user.lastLoginAt
```

Comment:
We should add `expect(user.lastLoginAt).not.to.be.undefined` before this -- no need to handle the case where `lastLoginAt` is not set. You can use `new Date(user.lastLoginAt!)` to tell TypeScript it is not null if it complains.